pkg/controllers/resources/persistentvolumeclaims/syncer.go

```diff
@@ -35,10 +36,15 @@ const (
 )
 
 func New(ctx *synccontext.RegisterContext) (syncer.Object, error) {
+	storageClassesEnabled := ctx.Controllers["storageclasses"]
+	excludedAnnotations := []string{bindCompletedAnnotation, boundByControllerAnnotation, storageProvisionerAnnotation}
+	if !storageClassesEnabled && ctx.Options.EnableScheduler {
```

Comment:
I think it would also make sense to check for the `ctx.Controllers["legacy-storageclasses"]` here and in the translateUpdate(). When "legacy-storageclasses" is enabled the StorageClasses will by synced from the host to the vcluster.
But right now we can merge it as is to fix the most pressing problem.